csrc/aes_gcm.cpp

```diff
@@ -28,37 +29,9 @@
 
 using namespace AmazonCorrettoCryptoProvider;
 
-class EVP_CIPHER_CTX_auto : public EVP_CIPHER_CTX {
-    public:
-        EVP_CIPHER_CTX_auto() {
-            EVP_CIPHER_CTX_init(this);
-        }
-        ~EVP_CIPHER_CTX_auto() {
-            EVP_CIPHER_CTX_cleanup(this);
-        }
-};
-
-static volatile long leak_tracking_enabled = 0;
-static volatile long active_instances = 0;
-
-struct native_context : public EVP_CIPHER_CTX {
-    public:
-        // True if this was counted to the leak tracking counters
-        bool leak_tracking;
-
-        native_context() {
-            leak_tracking = leak_tracking_enabled;
-            if (unlikely(leak_tracking)) __sync_add_and_fetch(&active_instances, 1);
-        }
-
-        ~native_context() {
-            if (unlikely(leak_tracking)) __sync_add_and_fetch(&active_instances, -1);
-        }
-};
-
 static void initContext(
   raii_env &env,
-  EVP_CIPHER_CTX *ctx,
+  raii_cipher_ctx &ctx,
```

Comment:
Q: What is the `raii_cipher_ctx` used for and how does it relate to the  `EVP_CIPHER_CTX`? I understand that this replaces the native_ctx used earlier.

Reply:
Found the answer to this question in this [class raii_cipher_ctx definition](https://github.com/corretto/amazon-corretto-crypto-provider/pull/60/files#diff-cc4e15e2836daff9a013fc62f827643dR93). 

Resolving this conversation.